src/Agent.Sdk/Util/VssUtil.cs

```diff
@@ -75,7 +99,7 @@ public static void InitializeVssClientSettings(ProductInfoHeaderValue additional
             settings.AcceptLanguages.Remove(CultureInfo.InvariantCulture);
 
             // Setting `ServerCertificateCustomValidation` to able to capture SSL data for diagnostic
-            if (trace != null)
+            if (trace != null && !PlatformUtil.RunningOnMacOS && IsCustomServerCertificateValidationSupported)
```

Comment:
```suggestion
            if (trace != null && IsCustomServerCertificateValidationSupported)
```
Even though the error messages say that custom callbacks are not supported on MacOS, I think if we have such simple check it doesn't hurt to try it.